apps/managedidentity/managedidentity_test.go

```diff
@@ -12,15 +12,16 @@ import (
 	"time"
```

Comment:
Could you add some tests where you can verify the whole "1 cache per process" goal? i.e. create 2 clients, get a token from client1 (source should be IdP), get a token from client2 (source should be cache)

Reply:
I actually create a second client `secondFakeClient` which after every test and gets the token and checks if it is from cache